modules/wallet/seed_test.go

```diff
@@ -278,57 +278,71 @@ func TestSweepSeedFunds(t *testing.T) {
 	// wallet. This wallet should have a siafund balance.
 	//
 	// TODO: when proper seed loading is implemented, this will be unnecessary.
-	w, err := New(wt.cs, wt.tpool, wt.wallet.persistDir)
+	wt.wallet, err = New(wt.cs, wt.tpool, wt.wallet.persistDir)
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer w.Close()
 	// reset the changeID
-	resetChangeID(w)
-	err = w.Unlock(wt.walletMasterKey)
+	resetChangeID(wt.wallet)
+	err = wt.wallet.Unlock(wt.walletMasterKey)
 	if err != nil {
 		t.Fatal(err)
 	}
-	_, siafundBal, _ := w.ConfirmedBalance()
+	_, siafundBal, _ := wt.wallet.ConfirmedBalance()
 	if siafundBal.Cmp(types.NewCurrency64(2000)) != 0 {
 		t.Error("expecting a siafund balance of 2000 from the 1of1 key")
 	}
+	// need to reset the miner as well, since it depends on the wallet
+	wt.miner, err = miner.New(wt.cs, wt.tpool, wt.wallet, wt.wallet.persistDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// send funds to ourself a few times
+	for i := 0; i < 10; i++ {
+		uc, err := wt.wallet.NextAddress()
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = wt.wallet.SendSiafunds(types.NewCurrency64(1), uc.UnlockHash())
+		if err != nil {
+			t.Fatal(err)
+		}
+		wt.addBlockNoPayout()
+	}
+	// send some funds to the void
+	_, err = wt.wallet.SendSiafunds(types.NewCurrency64(10), types.UnlockHash{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	wt.addBlockNoPayout()
```

Comment:
Would actually like to see this split into its own test. As far as work goes, you can do most of the heavy lifting via copy-paste.

I know that results in a ton of testing code, but it makes implementation easy, and more importantly you end up with tests that can have focused goals + purposes, which later on makes debugging and modification a million times easier.

In the old storage manager I had a bunch of really large tests, basically doing what you've just done above but taking it to a bigger extreme (some tests 300+ lines, some 900+ lines). Then when something broke, it was basically necessary to read the whole test to figure out where things had gone wrong, and then changing anything could potentially affect the entire rest of the test.

In host2 I've found it to be much more manageable to break them apart. The resulting files are massive, but when a test breaks its usually only 1 or 2 of them, and so far I've only had to spend more than a few minutes fixing it up just 1 time.

It's more important for the host because I'm doing a ton of dependency injection, but now it's just 1 dependency injection per test. But I think the same applies to the wallet nonetheless.

I think we will prefer maintaining these tests when there's tons of duplicate code but each test individually is just large enough to accomplish a single goal.